test/test.js

```diff
@@ -426,6 +426,39 @@ suite('include-fragment-element', function() {
     })
   })
 
+  test('loading events fire in guaranteed order', function() {
+    const elem = document.createElement('include-fragment')
+    const order = []
+    const connected = []
+    const events = [
+      when(elem, 'loadend').then(() => {
+        order.push('loadend')
+        connected.push(elem.isConnected)
+      }),
+      when(elem, 'load').then(() => {
+        order.push('load')
+        connected.push(elem.isConnected)
+      }),
+      when(elem, 'loadstart').then(() => {
+        order.push('loadstart')
+        connected.push(elem.isConnected)
+      })
+    ]
+    elem.src = '/hello'
+
+    // Emulate some kind of timer clamping
+    const originalSetTimeout = window.setTimeout
+    let i = 60
+    window.setTimeout = (fn, ms, ...rest) => originalSetTimeout.call(window, fn, ms + (i -= 20), ...rest)
+
+    document.body.appendChild(elem)
+    return Promise.all(events).then(() => {
+      window.setTimeout = originalSetTimeout
```

Comment:
Can we move the timer clamping emulation and restoration into `before` and `after` hooks so that we don't end up with an emulated `setTimeout` if the tests fail?